salt,build: Add version label on static manifest deployed

TeddyAndrieux opened the pull request:
**Component**:

'salt', 'build'
<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

Add version in static manifest
Use `dest_version` on `deploy_node` in downgrade orchestrate

**Summary**:

For upgrade and downgrade we want to know from which version we come.
See dde1aa163c943fc68e0b6118e9a7156712ddee8f for more detail

bert-e commented:
# Hello teddyandrieux,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/store-version-in-static-manifest` has diverged from
`development/2.4` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.4` into `improvement/store-version-in-static-manifest`
* Rebase `improvement/store-version-in-static-manifest` onto `origin/development/2.4`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/store-version-in-static-manifest` into
`development/2.0`
* [w/2.1/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/store-version-in-static-manifest) will be merged into `development/2.1` 
* [w/2.2/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/store-version-in-static-manifest) will be merged into `development/2.2` 
* [w/2.3/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/improvement/store-version-in-static-manifest) will be merged into `development/2.3` 
* [w/2.4/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/improvement/store-version-in-static-manifest) will be merged into `development/2.4` 
* [w/2.5/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/store-version-in-static-manifest) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

NicolasT reviewed (commented):
Instead of using build-time templating: the manifest YAMLs are (or could be) Jinja templates, and rendered through `file.managed`, splicing in whatever version string we want at that time...

In general, I think we should use Salt based rendering instead of build-time 'templating' where possible. The rationale: mainly consistency.

Makes sense?

TeddyAndrieux commented:
> Instead of using build-time templating: the manifest YAMLs are (or could be) Jinja templates, and rendered through `file.managed`, splicing in whatever version string we want at that time...
> 
> In general, I think we should use Salt based rendering instead of build-time 'templating' where possible. The rationale: mainly consistency.
> 
> Makes sense?

Hmm a user can put what ever he want in saltenv but since we use saltenv for k8s object it make sense to use saltenv here also. I will change this

TeddyAndrieux committed:
salt,build: Add version label on static manifest deployed

For upgrade and downgrade we want to know from which version we come.
See dde1aa1 for more detail

bert-e commented:
# History mismatch



Merge commit #4bed469ed63761eb542a852773e6edeb9dcf7f04 on the integration branch
`w/2.1/improvement/store-version-in-static-manifest` is merging a branch which is neither the current
branch `improvement/store-version-in-static-manifest` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/store-version-in-static-manifest` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/store-version-in-static-manifest` into
`development/2.0`
* [w/2.1/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/store-version-in-static-manifest) will be merged into `development/2.1` 
* [w/2.2/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/store-version-in-static-manifest) will be merged into `development/2.2` 
* [w/2.3/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/improvement/store-version-in-static-manifest) will be merged into `development/2.3` 
* [w/2.4/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/improvement/store-version-in-static-manifest) will be merged into `development/2.4` 
* [w/2.5/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/store-version-in-static-manifest) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

TeddyAndrieux committed:
salt: Fix `sync_all` from etcd orchestrate

In etcd orchestrate the `saltutil.sync_all` was wrong as we give a
`saltenv` directly to `salt.function` instead of `kwarg` for
`saltutil.sync_all`

bert-e commented:
# History mismatch



Merge commit #766806ae3d55b71f3c6f5d390d4b13873e319cf0 on the integration branch
`w/2.1/improvement/store-version-in-static-manifest` is merging a branch which is neither the current
branch `improvement/store-version-in-static-manifest` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/store-version-in-static-manifest` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/store-version-in-static-manifest` into
`development/2.0`
* [w/2.1/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/store-version-in-static-manifest) will be merged into `development/2.1` 
* [w/2.2/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/store-version-in-static-manifest) will be merged into `development/2.2` 
* [w/2.3/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/improvement/store-version-in-static-manifest) will be merged into `development/2.3` 
* [w/2.4/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/improvement/store-version-in-static-manifest) will be merged into `development/2.4` 
* [w/2.5/improvement/store-version-in-static-manifest](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/store-version-in-static-manifest) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

TeddyAndrieux commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **approve**

TeddyAndrieux committed:
salt: Use `deploy_node` of `dest_version` in downgrade

The downgrade orchestrate of the `current version` is called with a
salt-master already in the `destination version` so the orchestrate need
to be backward compatible.
Two choice:
1- we call the `deploy_node` of the `current version` so the
   `deploy_node` need also to be backward compatible
2- we call the `deploy_node` of the `destination version` so if we want
   to do specific stuff during downgrade we cannot add them directly in
   `deploy_node`

The first solution is rejected as we don't want to have backward
compatibility in an orchestrate used to deploy new nodes, so second
solution was chosen here

gdemonet reviewed (approved):
LGTM

TeddyAndrieux commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/c020298e5f39c0b77aee0ed702194eb9117b2cf0) did not succeed in branch w/2.1/improvement/store-version-in-static-manifest.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/5aa701756b2154332e4fb0f630bfee36786f895a) did not succeed in branch w/2.5/improvement/store-version-in-static-manifest.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`

* :heavy_check_mark: `development/2.3`

* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`

* :heavy_check_mark: `development/2.3`

* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue None.

Goodbye teddyandrieux.
